org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/navigation/FindImplementersHandler.java

```diff
@@ -0,0 +1,147 @@
+import com.google.gson.GsonBuilder;
+import java.util.ArrayList;
+import java.util.List;
+import org.eclipse.che.jdt.ls.extension.api.dto.ImplementersResponse;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.ITypeHierarchy;
+import org.eclipse.jdt.core.ITypeRoot;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.internal.corext.util.MethodOverrideTester;
+import org.eclipse.jdt.ls.core.internal.JDTUtils;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.jdt.ls.core.internal.ResourceUtils;
+import org.eclipse.jdt.ls.core.internal.handlers.DocumentSymbolHandler;
+import org.eclipse.jdt.ls.core.internal.hover.JavaElementLabels;
+import org.eclipse.lsp4j.Location;
+import org.eclipse.lsp4j.SymbolInformation;
+import org.eclipse.lsp4j.TextDocumentPositionParams;
+import org.eclipse.lsp4j.jsonrpc.json.adapters.CollectionTypeAdapterFactory;
+import org.eclipse.lsp4j.jsonrpc.json.adapters.EitherTypeAdapterFactory;
+import org.eclipse.lsp4j.jsonrpc.json.adapters.EnumTypeAdapterFactory;
+
+/**
+ * Command to find implementers of a type or a method
+ *
+ * @author dbocharo
+ */
+public class FindImplementersHandler {
+  private static final Gson gson =
+      new GsonBuilder()
+          .registerTypeAdapterFactory(new CollectionTypeAdapterFactory())
+          .registerTypeAdapterFactory(new EitherTypeAdapterFactory())
+          .registerTypeAdapterFactory(new EnumTypeAdapterFactory())
+          .create();
+
+  /**
+   * Finds implementers for an element defined by position
+   *
+   * @param parameters first parameter must be of type {@link TextDocumentPositionParams}, which
+   *     defines a position of an element which search implementers for
+   * @param pm a progress monitor
+   * @return an object of type {@link ImplementersResponse}, which contains description of a
+   *     searched element and implementers
+   */
+  @SuppressWarnings("restriction")
+  public static Object getImplementers(List<Object> parameters, IProgressMonitor pm) {
+    TextDocumentPositionParams param =
+        gson.fromJson(gson.toJson(parameters.get(0)), TextDocumentPositionParams.class);
+
+    ITypeRoot typeRoot = JDTUtils.resolveTypeRoot(param.getTextDocument().getUri());
+
+    ImplementersResponse implementersResponse = new ImplementersResponse();
+    List<SymbolInformation> implementers = new ArrayList<>();
+    implementersResponse.setImplementers(implementers);
+    try {
+      IJavaElement elementToSearch =
+          JDTUtils.findElementAtSelection(
+              typeRoot,
+              param.getPosition().getLine(),
+              param.getPosition().getCharacter(),
+              JavaLanguageServerPlugin.getPreferencesManager(),
+              pm);
+
+      switch (elementToSearch.getElementType()) {
+        case IJavaElement.TYPE:
```

Comment:
Do we want to restrict this to interfaces or are classes fine, too?

Reply:
Classes are fine too. Just tried to run it for an abstract class. Works fine - so no need in limiting it to interfaces. Although the word "implementation" is not very suitable for classes. "Extension" is better.